oshdb/src/test/java/org/heigit/ohsome/oshdb/contribution/ContributionsRelationTest.java

```diff
@@ -0,0 +1,315 @@
+            node(1, 1000, 101, 1, tags(), 0, 0)));
+
+    var versions = relations(1,
+        relation(1, 1000, 101, 1, tags(), mems(n(1, 0))));
+
+    var osh = osh(versions, nodes, emptyList());
+    var contribs = Contributions.of(osh);
+
+    assertTrue(contribs.hasNext());
+    var contrib = contribs.next();
+    assertContrib(1000, 101, 1, versions.get(0), contrib);
+    assertEquals(EnumSet.of(CREATION), contrib.getTypes());
+
+    assertFalse(contribs.hasNext());
+  }
+
+  @Test
+  void testNoMembers() {
+    var versions = relations(1,
+        relation(1, 1000, 101, 1, tags(), mems(n(1, 0))));
+
+    var osh = osh(versions, emptyList(), emptyList());
+    var contribs = Contributions.of(osh);
+
+    assertTrue(contribs.hasNext());
+    var contrib = contribs.next();
+    assertContrib(1000, 101, 1, versions.get(0), contrib);
+    assertEquals(EnumSet.of(CREATION), contrib.getTypes());
+
+    assertFalse(contribs.hasNext());
+  }
+
+  @Test
+  void testMissingMemberChanges() {
+    var nodes = List.of(
+        osh(1,
+            node(1, 1000, 101, 1, tags(), 0, 0)));
+    var ways = List.of(
+        osh(1, List.of(
+            osh(1,
+                node(1, 1000, 101, 1, tags(), 0, 0))),
+            way(1, 1000, 101, 1, tags(), mems(1))));
+
+    var versions = relations(1,
+        relation(1, 1000, 101, 1, tags(), mems(w(1, 0), r(1, 0))));
+
+    var osh = osh(versions, nodes, ways);
+    var contribs = Contributions.of(osh);
+
+    assertTrue(contribs.hasNext());
+    var contrib = contribs.next();
+    assertContrib(1000, 101, 1, versions.get(0), contrib);
+    assertEquals(EnumSet.of(CREATION), contrib.getTypes());
+
+    assertFalse(contribs.hasNext());
+  }
+
+  @Test
+  void testDifferentWayAndNode() {
+    var nodes = List.of(
+        osh(1,
+            node(1, 1000, 101, 1, tags(), 0, 0)),
+        osh(2,
+            node(1, 1000, 101, 1, tags(), 0, 0)),
+        osh(3,
+            node(1, 1000, 101, 1, tags(), 0, 0)));
+    var ways = List.of(
+        osh(1, List.of(
+            osh(1,
+                node(1, 1000, 101, 1, tags(), 0, 0))),
+            way(1, 1000, 101, 1, tags(), mems(1))),
+        osh(2, List.of(
+            osh(1,
+                node(1, 1000, 101, 1, tags(), 0, 0))),
+            way(1, 1000, 101, 1, tags(), mems(1))),
+        osh(3, List.of(
+            osh(1,
+                node(1, 1000, 101, 1, tags(), 0, 0))),
+            way(1, 1000, 101, 1, tags(), mems(1))));
+
+    var versions = relations(1,
+        relation(2, 2000, 202, 2, tags(),
+            mems(w(1, 0), n(1, 0), w(2, 0), n(2, 0), w(1, 0), n(1, 0))),
+        relation(1, 1000, 101, 1, tags(), mems(w(1, 0), n(1, 0), w(3, 0), n(3, 0))));
+
+    var osh = osh(versions, nodes, ways);
+    var contribs = Contributions.of(osh);
+
+    assertTrue(contribs.hasNext());
+    var contrib = contribs.next();
+    assertContrib(2000, 202, 2, versions.get(0), contrib);
+    assertEquals(EnumSet.of(MEMBER_CHANGE), contrib.getTypes());
```

Comment:
Do we need to check the members here (and maybe in the older contributions as well)?